kernel_tuner/cache/json.py

```diff
@@ -1,14 +1,18 @@
-from typing import TypedDict, NotRequired, Any
+from __future__ import annotations
+from typing import TypedDict, Any
 
 
 _non_identifier_keys = TypedDict("CacheEntryJSON", {
-    "GFLOP/s": NotRequired[float]
+    "GFLOP/s": float
 })
 
 
-class CacheLineJSON(TypedDict, _non_identifier_keys):
+class CacheLineOptionalJSON(TypedDict, _non_identifier_keys, total=False):
```

Comment:
Waar is `total` precies voor?

Reply:
De Kernel Tuner library support officieel Python 3.8 en daarom werk ik met Python 3.8. In Python 3.8, is `NotRequired` nog niet gedefinieerd, dus daarom gebruik ik `total=false`, zodat de properties niet allemaal required zijn, want dat is al wel gedefinieerd in Python 3.8.

We kunnen dus nog toevoegen aan onze DoD dat de code moet werken met Python 3.8.

Reply:
Ik vind dat een goed idee, om dat in DoD te zetten dat de code moet werken met Python 3.8